src/components/loadable-item.cjsx

```diff
@@ -28,57 +28,75 @@ module.exports = React.createClass
   getDefaultProps: ->
     bindEvent: 'change'
 
-  componentDidMount: -> @reload({})
-  componentDidUpdate: (oldProps) -> @reload(oldProps)
+  componentWillMount: ->
+    {id, store, options} = @props
+    @load(id, options) if id? and not store.isNew(id, options)
 
-  reload: (oldProps) ->
-    {id, store, load, actions, options} = @props
+  componentWillReceiveProps: (nextProps) ->
+    @reload(@props, nextProps)
+
+  arePropsSame: (prevProps, nextProps) ->
+    {id, store, load, actions, options} = nextProps
+
+    prevProps.id is id and
+      prevProps.store is store and
+      prevProps.actions is actions and
+      prevProps.load is load and
+      _.isEqual(prevProps.options, options)
+
```

Comment:
I'm a little confused by the explicit `xxx is yyy` comparisons.  Why not just use `_.isEqual(prevProps, nextProps)` here?  That'd be small enough that you could just do it inline vs having the `arePropsSame` method. 

I'm pretty sure there's a good reason since you went to the trouble though.  Maybe a good spot for an explanation comment 😁